Student/Student/Assignments/AssignmentDetails/AssignmentDetailsViewController.swift

```diff
@@ -247,6 +256,45 @@ class AssignmentDetailsViewController: UIViewController, AssignmentDetailsViewPr
             submitAssignmentButton.alpha = 0
         }
     }
+
+    // MARK: - Show / Hide Sections
+
+    func showLocked(assignment: Assignment) {
+        let lockStatus = assignment.lockStatus
+
+        switch lockStatus {
+        case .unlocked:
+            lockedSection?.isHidden = true
+        case .before:
+            lockedIconContainerView.isHidden = false
+            dueSection?.isHidden = true
+            lockedSection?.isHidden = false
+            fileTypesSection?.isHidden = true
+            submissionTypesSection?.isHidden = true
+            gradeSection?.isHidden = true
+            submissionButtonSection?.isHidden = true
+            showDescription(false)
+            submitAssignmentButton.isHidden = true
+        case .after:
+            lockedSection?.isHidden = false
+            submitAssignmentButton.isHidden = true
```

Comment:
Should this case also update all the views in `.before` in the scenario where the `lockStatus` changes in a pull to refresh?

Reply:
yeah, good point